src/main/java/bio/terra/service/dataset/AssetSpecification.java

```diff
@@ -73,4 +74,26 @@ public AssetTable getAssetTableByName(String tableName) {
         .findFirst()
         .orElseThrow();
   }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
We do this, but we really shouldn't have `hashCode()` and `equals()` on a class with mutable members.

I would either change the code that's doing `equals()` on this to use name or ID to match on, or to make all fields in this class `final`